src/functions/indPolyhedralQPDAS.jl

```diff
@@ -0,0 +1,146 @@
+# IndPolyhedral: QPDAS implementation
+
+import QPDAS
+
+"""
+**Indicator of a polyhedral set**
+
+    IndPolyhedralQPDAS(A, C, b, d)
+
+```math
+S = \\{x : \\langle A, x \\rangle = b\\ ∧ \\langle C, x \\rangle \\le b\\}.
+```
+"""
+struct IndPolyhedralQPDAS{R<:Real, MT<:AbstractMatrix{R}, VT<:AbstractVector{R}, QP<:QPDAS.QuadraticProgram} <: IndPolyhedral
+    A::MT
+    b::VT
+    C::MT
+    d::VT
+    z::VT
+    qp::QP
+    first_prox::Ref{Bool}
+    function IndPolyhedralQPDAS{R}(A::MT, b::VT, C::MT, d::VT) where {R<:Real, MT<:AbstractMatrix{R}, VT<:AbstractVector{R}, QP<:QPDAS.QuadraticProgram}
+        @assert size(A,1) == size(b,1)
+        qp = QPDAS.QuadraticProgram(A, b, C, d, smartstart=false)
+        new{R, MT, VT, typeof(qp)}(A, b, C, d, zeros(R,size(A,2)), qp, Ref(true))
+    end
+end
+
+# properties
+
+is_prox_accurate(::IndPolyhedralQPDAS) = true
+
+# constructors
+
+function IndPolyhedralQPDAS(
+    l::AbstractVector{R}, A::AbstractMatrix{R}, u::AbstractVector{R}
+) where R
+    if !all(l .<= u)
+        error("function is improper (are some bounds inverted?)")
+    end
+    eqinds = (l .== u) .& .!isnothing.(l)
+    Aeq = A[eqinds,:]
+    beq = l[eqinds]
+
+    _islower(l::T) where T =
+        l != typemin(T) && !isnan(l) && !isnothing(l)
+    _isupper(u::T) where T =
+        u != typemax(T) && !isnan(u) && !isnothing(u)
+
+    lower = _islower.(l) .& (.!eqinds)
+    upper = _isupper.(u) .& (.!eqinds)
+
+    lower_only = lower .& (.! upper)
+    upper_only = upper .& (.! lower)
+    upper_and_lower = upper .& lower
+
+
+    Cieq = [-A[lower_only, :];
+             A[upper_only, :];
+            -A[upper_and_lower, :];
+             A[upper_and_lower, :] ]
+    dieq = [-l[lower_only];
+             u[upper_only];
+            -l[upper_and_lower];
+             u[upper_and_lower] ]
+
+    IndPolyhedralQPDAS{R}(Aeq, beq, Cieq, dieq)
+end
+
+IndPolyhedralQPDAS(
+    l::AbstractVector{R}, A::AbstractMatrix{R}, u::AbstractVector{R},
+    xmin::AbstractVector{R}, xmax::AbstractVector{R}
+) where R =
+    IndPolyhedralQPDAS([l; xmin], [A; I], [u; xmax])
+
+IndPolyhedralQPDAS(
+    l::AbstractVector{R}, A::AbstractMatrix{R}, args...
+) where R =
+    IndPolyhedralQPDAS(
+        l, A, R(Inf).*ones(R, size(A, 1)), args...
+    )
+
+IndPolyhedralQPDAS(
+    A::AbstractMatrix{R}, u::AbstractVector{R}, args...
+) where R =
+    IndPolyhedralQPDAS(
+        R(-Inf).*ones(R, size(A, 1)), A, u, args...
+    )
+
+# function evaluation
+
+function (f::IndPolyhedralQPDAS{R})(x::AbstractVector{R}) where R
+    Ax = f.A * x
+    Cx = f.C * x
+    return all(Ax .<= f.b .& Cx .<= f.d) ? R(0) : Inf
```

Comment:
Is it `Ax .== f.b` here? Or rather `isapprox.(Ax, f.b)`.

If this was a bug and no test failed, maybe a test is missing

Reply:
Bug, will fix and add test.